File: bitcode_android/Day03/app/src/main/java/in/kaligotla/day03/MainActivity.java
package in.kaligotla.day03;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.TextView;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {
    ImageView imgView;
    TextView welcomeTxtView;
    EditText edtUserName;
    Button btnSubmit;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initViews();
        bindData();
        setListeners();
    }

    private void initViews() {
        imgView = findViewById(R.id.imgView);
        welcomeTxtView = findViewById(R.id.welcomeTxtView);
        edtUserName = findViewById(R.id.edtUserName);
        btnSubmit = findViewById(R.id.btnSubmit);
    }

    private void bindData() {
        imgView.setImageResource(R.drawable.ic_launcher_background);
        welcomeTxtView.setText("Welcome to BitCode Technologies");
    }

    private void setListeners() {
        btnSubmit.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                welcomeTxtView.setText("Welcome "+edtUserName.getText());
            }
        });
    }

    /*
    Resources is a class in Android, cannot create object, use "getResources()".
    Types of resources -
        * Drawable & MipMap -> (images)
        * Layout -> (UI, xml files)
        * values -> (Colors, Strings, integers, array, typed-array, dimens.xml (dimensions), styles, booleans)
        * xml ->
    */
}

File: bitcode_android/NavigationDemo/app/src/main/java/in/kaligotla/navigationdemo/MainActivity.kt
package `in`.kaligotla.navigationdemo

import android.content.Intent
import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import `in`.kaligotla.navigationdemo.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var activityMainBinding: ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        activityMainBinding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(activityMainBinding.root)
        activityMainBinding.next.setOnClickListener {
            val intent: Intent = Intent(this@MainActivity, SecondActivity::class.java)
            intent.putExtra("key", "Value")
            startActivityForResult(intent, 1)
        }
    }
}